### PR TITLE
clearvpn 3.0.0 (new cask)

### PR DESCRIPTION
**Pull request by rternavskyi**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by razvanazamfirei** [comment]
Thank you for opening this! I made some changes. The livecheck strategy is a bit awkward, but we prefer versioned urls when feasible.

**Comment by krehel** [comment]
Could we go simpler on this?  Not sure if we need the `shortVersionString` - this strips out pretty much any excess in the regex, but I don't want to be so simple it causes issues down the line.

```
  livecheck do
    url "https://updates.devmate.com/com.macpaw.clearvpn.macos-site-ver.xml"
    regex(%r{/(\d+)/ClearVPN}i)
    strategy :sparkle do |item, regex|
      match = item.url.match(regex)
      next if match.blank?

      "#{item.title},#{match[1]}"
    end
  end
 ```

Alternate regex that also works with using the `.zip` to help maybe reduce any false positives -

```
regex(%r{/(\d+)/ClearVPN[._-](?:\d+\.+\d+)+\.zip}i)
```

**Comment by rternavskyi** [comment]
@razvanazamfirei Why is there such a complication of livecheck? Here is a script for another program from the same publisher.
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/c/cleanmymac.rb

**Comment by bevanjkay** [comment]
@rternavskyi We need to collect additional version parts to assemble the `url` dynamically. The cask you linked has a simpler version format upstream.

**Comment by razvanazamfirei** [comment]
@krehel, that looks good to me. I wanted to keep the canonical version, but am fine with us dropping it.

File: Casks/c/clearvpn.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+cask "clearvpn" do
+  version "3.0.0,202403.21.1808,1711062454"
+  sha256 "8b20eba3df6e0109aee4f041db52007d0c5672b78599ecdd9e3d1140b123a835"
+
+  url "https://dl.devmate.com/com.macpaw.clearvpn.macos-site-ver/#{version.csv.second}/#{version.csv.third}/ClearVPN-#{version.csv.second}.zip",
+      verified: "dl.devmate.com/"
+  name "ClearVPN"
+  desc "One-tap VPN for secure online journeys"
```
Comment (bevanjkay) [inline Casks/c/clearvpn.rb:8]:
```suggestion
  desc "VPN client"
```

**Review by bevanjkay — CHANGES_REQUESTED** [review]
Looks good to me, just a minor nit on the `desc`.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**